packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -141,16 +144,50 @@ public class AmplifyStorageOperations {
         return itemAsDictionary
     }
     
-    private static func prepareError(flutterResult: @escaping FlutterResult, msg: String, localizedError: String?, recoverySuggestion: String?) {
-        var errorDetails: [String: Any] = [:]
-        var platformErrorDetails: [String: String] = [:]
-        
-        platformErrorDetails["platform"] = "iOS"
-        platformErrorDetails["localalizedError"] = localizedError
-        platformErrorDetails["recoverySuggestion"] = recoverySuggestion
-        
-        errorDetails["PLATFORM_EXCEPTIONS"] = platformErrorDetails
+    private static func prepareError(flutterResult: @escaping FlutterResult, error: Error) {
+        var message: String = ErrorMessages.missingExceptionMessage
+        var recoverySuggestion: String = ErrorMessages.missingRecoverySuggestion
+        var details: Dictionary<String, String>
         
-        flutterResult(FlutterError(code: "AmplifyException", message: msg, details: errorDetails))
+        if (error is StorageError) {
+            details = createSerializedError(storageError: error as! StorageError);
+        } else if (error is AmplifyFlutterValidationException) {
+            let e = error as! AmplifyFlutterValidationException
```

Comment:
nit:`if-let` will probably avoid these nasty force casts

Reply:
I don't think I need to check this anymore... the validation exception is just a plain custom error.